integration/client/daemon.go

```diff
@@ -59,21 +60,30 @@ func (d *daemon) waitForStart(ctx context.Context) (*Client, error) {
 		client  *Client
 		serving bool
 		err     error
+		ticker  = time.NewTicker(500 * time.Millisecond)
 	)
-
-	client, err = New(d.addr)
-	if err != nil {
-		return nil, err
-	}
-	serving, err = client.IsServing(ctx)
-	if !serving {
-		client.Close()
-		if err == nil {
-			err = errors.New("connection was successful but service is not available")
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			client, err = New(d.addr)
+			if err != nil {
+				continue
+			}
```

Comment:
Haven't looked closely in the whole code, but this one felt a bit odd; is it needed to create a new client in the loop (so construct a new client for every request)?

Reply:
Creating a new client can sometimes fast fail, as we saw in here #7422. This change addresses that by retrying the client connection on failure at least until  the context deadline has exceeded.

Reply:
Does that mean that `New(d.addr)` already makes a connection? That is a bit surprising (I wouldn't immediately expect that constructing a client requires a connection); if that's the case, I feel like that's something we need to look into.